### PR TITLE
fix: allow `visible_labels=None` & disallow `fields`/`questions` with the same name

### LINKED ISSUE #3034: Disallow `fields` and/or `questions` with same `name`
**Describe the bug**

When defining a `FeedbackDataset` the following use-cases shouldn't be accepted and an exception should be raised in the client-side:

```python
import argilla as rg

ds = rg.FeedbackDataset(fields=[rg.TextField(name="text"), rg.TextField(name="text")], questions=...)
ds = rg.FeedbackDataset(fields=..., questions=[rg.TextQuestion(name="text"), rg.TextQuestion(name="text")])
ds = rg.FeedbackDataset(fields=[rg.TextField(name="text")], questions=[rg.TextQuestion(name="text")])
```

So on, we should handle that all the names across `fields` and `questions` are unique.

**Expected behavior**

A `ValueError` to be raised if the names across `fields` and `questions` are not unique.

**Environment (please complete the following information):**
 - OS [e.g. iOS]: N/A
 - Browser [e.g. chrome, safari]: N/A
 - Argilla Version [e.g. 1.0.0]: 1.8.0
 - ElasticSearch Version [e.g. 7.10.2]: N/A
 - Docker Image (optional) [e.g. argilla:v1.0.0]: argilla:v1.8.0

### LINKED ISSUE #3123: Duplicated question error message upon creating dataset
**Is your feature request related to a problem? Please describe.**
If I create in a dataset a question that already exists in that dataset, I get an error message when I try to push the dataset. I would like to have this validation & error message when I create the dataset locally, not when I push it.

### PR DESCRIPTION
**Pull request by alvarobartt**
# Description

This PR solves two minor fixes reported before the 1.9.0 release related to the validations to make sure that:

* `visible_labels` in `LabelQuestion` and/or `MultiLabelQuestion` can be set to `None` so that all the labels show in the Argilla UI
* The fields and questions of a `FeedbackDataset` cannot have the same name, otherwise, a `ValueError` should be raised

Closes #3034, #3123

**Type of change**

- [X] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [X] Add unit tests for `visible_labels=None` to make sure it's kept and not replaced by default 20
- [X] Add unit tests for `FeedbackDataset` init with duplicated field and question names

**Checklist**

- [X] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,11 @@ These are the section headers that we use:
 - Database setup for unit tests. Now the unit tests use a different database than the one used by the local Argilla server (Closes [#2987]).
 - Updated `alembic` setup to be able to autogenerate revision/migration scripts using SQLAlchemy metadata from Argilla server models ([#3044])
 
+### Fixed
+
+- Disallow `fields` and `questions` in `FeedbackDataset` with the same name ([#3126]).
+- Keep `visible_labels=None` if `None` is specified in `LabelQuestion` or `MultiLabelQuestion`, otherwise, use default 20 ([#3126]).
+
```
Comment (frascuchon) [inline CHANGELOG.md:44]:
```suggestion
### Fixed

- Disallow `fields` and `questions` in `FeedbackDataset` with the same name ([#3126]).
- Keep `visible_labels=None` if `None` is specified in `LabelQuestion` or `MultiLabelQuestion`, otherwise, use default 20 ([#3126]).

[#3126]: https://github.com/argilla-io/argilla/pull/3126

```

**Review by frascuchon — APPROVED** [review]
Just a comment to show links properly in the CHANGELOG.md
